### PR TITLE
feat(components): add camel-observability-services

### PR DESCRIPTION
**Pull request by squakez**
With this PR we're adding a new component which serves as an adapter and helper to expose a set of opinionated observability services. This first MVP adds `camel-micrometer-prometheus` components and a default configuration for it. Once this is onboarded I'll proceed with creating the proper extension/starter for the other runtimes.

Iteratively we'll be adding into this adapter a few more components in order to expose other sensitive observability metrics (telemetry, health, ...). The main goal is to provide services required when running a microservice oriented architecture on cloud, but the component can be used outside cloud as well with the same meaning.

Closes CAMEL-21384

# Description

# Target

- [x] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [x] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [x] I checked that each commit in the pull request has a meaningful subject line and body.

- [x] I have run `mvn clean install -DskipTests` locally from root folder and I have committed all auto-generated changes.

### CODE REVIEW

**Comment by squakez** [comment]
@claudio4j @oscerd FYI. I am not sure if I've missed some configuration required to onboard the new component. Just let me know.

File: core/camel-main/src/main/java/org/apache/camel/main/BaseMainSupport.java [hunk 1]
```diff
@@ -145,7 +145,8 @@ public abstract class BaseMainSupport extends BaseService {
     protected final OrderedLocationProperties wildcardProperties = new OrderedLocationProperties();
     protected RoutesCollector routesCollector = new DefaultRoutesCollector();
     protected String propertyPlaceholderLocations;
-    protected String defaultPropertyPlaceholderLocation = MainConstants.DEFAULT_PROPERTY_PLACEHOLDER_LOCATION;
+    protected String defaultPropertyPlaceholderLocation
+            = MainConstants.DEFAULT_PROPERTY_PLACEHOLDER_LOCATION + "," + MainConstants.DEFAULT_OBSERVABILITY_SERVICES_PROPERTY_LOCATION ;
```
Comment (squakez) [inline core/camel-main/src/main/java/org/apache/camel/main/BaseMainSupport.java:149]: This is required in order to load the configuration properties when available on classpath.

File: core/camel-main/src/main/java/org/apache/camel/main/MainConstants.java [hunk 2]
```diff
@@ -19,6 +19,8 @@
 public final class MainConstants {
 
     public static final String DEFAULT_PROPERTY_PLACEHOLDER_LOCATION = "classpath:application.properties;optional=true";
+    public static final String DEFAULT_OBSERVABILITY_SERVICES_PROPERTY_LOCATION
+            = "classpath:observability-services.properties;optional=true";
```
Comment (davsclaus) [inline core/camel-main/src/main/java/org/apache/camel/main/MainConstants.java:23]: yes correct it should be marked as optional

**Review by gansheer — APPROVED** [review]

**Review by davsclaus — APPROVED** [review]

**Review by oscerd — APPROVED** [review]
LGTM, thanks.

**Pull request merged**